server/main.lua

```diff
@@ -41,6 +41,27 @@ QBX.UsableItems = {}
 local vehicleClasses
 local vehicleClassesPromise
 
+local currentSessionId = 1
+local sessionIdLock = false
+
+---Sets a unique sessionId statebag on the entity.
+---@param entity number
+---@return integer sessionId
+local function createSessionId(entity)
+    if sessionIdLock then
+        lib.waitFor(function()
```

Comment:
Is it wild to assume there may be multiple different instances wanting to create a session ID at the same time? In such a case they will all wait together, resolve together, and possibly receive the same session ID. Perhaps some kind of queue should be made instead? Or handle it with promises?

Reply:
Yeah you are correct

Reply:
Per offline discussion and testing, we're unable to create a scenario where two sessionIds are duplicate. Thus the conclusion is that this is either impossible in the five runtime, or so unlikely as not be worth handling.